### PR TITLE
Rimsec-Security patch

### PR DESCRIPTION
**Pull request by fors4ken**
Mod link - https://steamcommunity.com/sharedfiles/filedetails/?id=2323762086

The droids and turrets are strictly better than vanilla counter parts. This is because nothing in this mod is spammable. You need special components to craft each and every item so I've considered that before patching.  The special components are sold by a trader that ONLY visits 3 times a year. They are expensive as well. But still the mod should be reasonably balanced. :) I'm open to balancing suggestions 

## Testing

Check tests you have performed:
- [x] Compiles without warnings
- [x] Game runs without errors
- [x] (For compatibility patches) ...with and without patched mod loaded
- [x] Playtested a colony (specify how long) - 2hrs

### CODE REVIEW

**Comment by Breadbox2k19** [comment]
Took a quick glance through your xml(haven't gone through it in-game yet).
I strongly recommend creating a new projectile(as the mod itself contains projectile texture) for the weapons instead of just using 6x24/6x18/8mm charge projectile, you can make the same ammo fire different projectiles by creating a new ammoset.

Alternatively, you can also create a universal Rimsec ammo used by all the rimsec weapons, you can make the different weapon consume ammo at a different rate by adding 					<ammoConsumedPerShotCount>2</ammoConsumedPerShotCount>
to the weapon <Properties>.

For balancing purpose I would also recommend adjusting down the health scale of the robots, the sentinel from 3 to 2.5 and the enforcer from 2 to 1.6 now that they have strong armour.

**Comment by Breadbox2k19** [comment]
Another thing, while I do not believe they appear in any raid, their pawnkind should still be patched to spawn with proper ammo+combat power adjusted up to reflect the appropriate power of a heavily armed and armoured pawn.

**Comment by fors4ken** [comment]
> Took a quick glance through your xml(haven't gone through it in-game yet).
> I strongly recommend creating a new projectile(as the mod itself contains projectile texture) for the weapons instead of just using 6x24/6x18/8mm charge projectile, you can make the same ammo fire different projectiles by creating a new ammoset.
> 
> Alternatively, you can also create a universal Rimsec ammo used by all the rimsec weapons, you can make the different weapon consume ammo at a different rate by adding 2
> to the weapon .
> 
> For balancing purpose I would also recommend adjusting down the health scale of the robots, the sentinel from 3 to 2.5 and the enforcer from 2 to 1.6 now that they have strong armour.

Good suggestion. I'll make these changes.

**Comment by fors4ken** [comment]
> Another thing, while I do not believe they appear in any raid, their pawnkind should still be patched to spawn with proper ammo+combat power adjusted up to reflect the appropriate power of a heavily armed and armoured pawn.

Yeah, they don't appear anywhere but I'll patch them anyways

**Comment by Breadbox2k19** [comment]
Oh and before I forget, remember to add ammo to the mod traders, especially ammo for modded weapons.

File: Patches/RimSec-Security/AlienBase.xml [hunk 1]
```diff
@@ -0,0 +1,266 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<Patch>
+	<Operation Class="PatchOperationSequence">
```
Comment (N7Huntsman) [inline Patches/RimSec-Security/AlienBase.xml:3]: What's up with the extra PatchOpSequence here?

Comment (fors4ken) [reply Patches/RimSec-Security/AlienBase.xml:3]: That's an oopsie. Will fix.

**Comment by fors4ken** [comment]
@Breadbox2k19 I've added charged ammo equivalents for SRS guns and turrets that uses the proj sprites from the mod. This of course wont clutter the fab bench with new ammo. The 8mm & 12mm railguns are unchanged since I'm thinking of creating entirely new ammunition for them. But for now the patch is complete from my side. Others are welcome to contribute if anything is remaining.

**Comment by Person243546** [comment]
I tested the turrets and they are too short (0.88m)

**Comment by fors4ken** [comment]
> 
> 
> I tested the turrets and they are too short (0.88m)

Should they be the same as walls ?

**Comment by Person243546** [comment]
Usually the fill percent of turrets in CE is 0.85 (1.49m in game)

**Review by Breadbox2k19 — APPROVED** [review]
Lookin good.

**Comment by clone2122** [comment]
In my testings bots can't use loadout system. Is it intended behavior?

**Pull request merged**